avocado/core/safeloader.py

```diff
@@ -30,12 +30,48 @@ class AvocadoModule(object):
     """
     Representation of a module that might contain avocado.Test tests
     """
-    __slots__ = 'path', 'test_import', 'mod_import'
+    __slots__ = 'path', 'test_import', 'mod_import', 'mod'
 
     def __init__(self, path, test_import=False, mod_import=False):
         self.path = path
         self.test_import = test_import
         self.mod_import = mod_import
+        if os.path.isdir(path):
+            self.path = os.path.join(path, "__init__.py")
+        else:
+            self.path = path
+        with open(self.path) as source_file:
+            self.mod = ast.parse(source_file.read(), self.path)
+
+    def iter_classes(self):
+        """
+        Iter through classes and keep track of imported avocado statements
+        """
+        for statement in self.mod.body:
+            # Looking for a 'from avocado import Test'
+            if (isinstance(statement, ast.ImportFrom) and
+                    statement.module == 'avocado'):
+
+                for name in statement.names:
+                    if name.name == 'Test':
+                        if name.asname is not None:
+                            self.test_import = name.asname
+                        else:
+                            self.test_import = name.name
+                        break
+
+            # Looking for a 'import avocado'
+            elif isinstance(statement, ast.Import):
+                for name in statement.names:
+                    if name.name == 'avocado':
+                        if name.asname is not None:
+                            self.mod_import = name.nasname
```

Comment:
This brings back the issue fixed here:

https://github.com/avocado-framework/avocado/pull/2773/commits/86f7d7717c8bb5cc25fbaaaf031bbdf1fddc05a6#diff-970dfe218a6d9b44e04b3b313be3078eR237

Reply:
yep, sorry, accidentally fixed it but the proper fix is included later along with selftest...